### PR TITLE
Adding argMax and argMin ops and convert_axis utility function.

### PR DESCRIPTION
**Pull request by kevinch-nv**
Also updated other OPS to use new getAxis function.

### CODE REVIEW

File: builtin_op_importers.cpp [hunk 1]
```diff
@@ -423,6 +423,66 @@ bool registerBuiltinOpImporter(std::string op,
...
+NodeImportResult argMinMaxHelper(IImporterContext* ctx,
+    const ::ONNX_NAMESPACE::NodeProto& node, std::vector<TensorOrWeights>& inputs, nvinfer1::TopKOperation op)
+{
+    ASSERT(inputs.at(0).is_tensor(), ErrorCode::kUNSUPPORTED_NODE);
+    nvinfer1::ITensor& tensor = inputs.at(0).tensor();
+    ASSERT(tensor.getType() != nvinfer1::DataType::kINT32, ErrorCode::kUNSUPPORTED_NODE);
+    // Get attributes.
+    OnnxAttrs attrs(node);
+    int keepdims = attrs.get("keepdims", 1);
+    int axis = attrs.get("axis", -1);
```
[the remark this answers is not in the document]
Comment (kevinch-nv) [reply builtin_op_importers.cpp]: Yes, fixed.

File: builtin_op_importers.cpp [hunk 2]
```diff
@@ -423,6 +423,66 @@ bool registerBuiltinOpImporter(std::string op,
...
+    ASSERT(axis != BATCH_DIM, ErrorCode::kUNSUPPORTED_NODE);
+
+    // Insert a TopK layer with k set to 1.
+    int nbDims = tensor.getDimensions().nbDims;
+    axis = getAxis(axis, nbDims);
+    ASSERT(axis >= 0 && axis < nbDims, ErrorCode::kUNSUPPORTED_NODE);
```
[the remark this answers is not in the document]
Comment (kevinch-nv) [reply builtin_op_importers.cpp]: Good idea, implemented.

File: onnx2trt_utils.hpp [hunk 3]
```diff
@@ -238,6 +238,20 @@ inline int div_ceil(int n, int d) {
   return (n - 1) / d + 1;
 }
 
+// Given an axis in ONNX and the rank of the TRT input tensor, returns an axis compatible with TRT.
+inline int getAxis(int axis, int nbDims)
```
[the remark this answers is not in the document]
Comment (kevinch-nv) [reply onnx2trt_utils.hpp]: changed to convert_axis to be in line with the other functions in onnx2trt_utils.hpp

File: builtin_op_importers.cpp [hunk 4]
```diff
@@ -423,6 +423,66 @@ bool registerBuiltinOpImporter(std::string op,
...
+        {
+            reshapeDims.d[i] = reshapeDims.d[i + 1];
+        }
+        nvinfer1::IShuffleLayer* squeezeLayer = ctx->network()->addShuffle(*indices);
+        squeezeLayer->setReshapeDimensions(reshapeDims);
+        return {{squeezeLayer->getOutput(0)}};
+    }
+}
+
+DEFINE_BUILTIN_OP_IMPORTER(ArgMax)
```
[the remark this answers is not in the document]
Comment (kevinch-nv) [reply builtin_op_importers.cpp]: Done.

File: onnx2trt_utils.hpp [hunk 5]
```diff
@@ -238,6 +238,35 @@ inline int div_ceil(int n, int d) {
   return (n - 1) / d + 1;
 }
 
+// Given an axis and TRT input tensor, modify and return the axis value
+// depending on whether or not the input is a TRT tensor.
+inline Status convert_axis(int& axis, int nbDims, bool isTensor)
+{
+  enum { BATCH_DIM = 0 };
```
Comment (benbarsdell) [inline onnx2trt_utils.hpp]: Never used

File: onnx2trt_utils.hpp [hunk 6]
```diff
@@ -238,6 +238,35 @@ inline int div_ceil(int n, int d) {
...
+  // If the input is a TRT tensor, no modification to axis is necessary
```
Comment (benbarsdell) [inline onnx2trt_utils.hpp]: My brain is mush at the moment, but this confuses me. Doesn't it still need to subtract 1 and check that it's not the batch dim?

File: onnx2trt_utils.hpp (hunk 5 above)
Comment (kevinch-nv) [reply onnx2trt_utils.hpp]: Removed

File: onnx2trt_utils.hpp (hunk 6 above)
Comment (kevinch-nv) [reply onnx2trt_utils.hpp]: Seems my brain was more mush :boom:. Good catch

**Pull request merged**